File: TCPServer/Scala/TCPClient.scala
//Includes
 import java.net.Socket;
 import java.io.InputStream;
 import java.io.OutputStream;

object TCPClient
{
 def main(args:Array[String])=
 {
  while(true)
  {
   var buffer=new Array[Byte](256)
   var message=System.console().readLine()                //Read message from console
   var client=new Socket("127.0.0.1",3721)                //Connect to server
   var output=client.getOutputStream()                    //Get client output stream
   var input=client.getInputStream()                      //Get client input stream
   output.write((message+"\n").getBytes())                //Write message to server
   var size=input.read(buffer)                            //Read response from server
   System.out.print(new String(buffer).substring(0,size)) //Write response to console
   client.close()                                         //Close client
  }
 }
}

File: TCPServer/C#/TCPClient.cs
//Includes
 using System;
 using System.Text;
 using System.Net;
 using System.Net.Sockets;

public class TCPClient
{
 public static void Main()
 {
  byte[] buffer = new byte[0x100];
  while(true)
  {
   string message=Console.ReadLine()+"\n";                                                      //Read message from console
   Socket client = new Socket(AddressFamily.InterNetwork, SocketType.Stream, ProtocolType.Tcp); //Create the client
   client.Connect(new IPEndPoint(IPAddress.Loopback, 3721));                                    //Connect to the address
   client.Send(Encoding.ASCII.GetBytes(message));                                               //Send the message to the server
   int size = client.Receive(buffer, 0, 255, 0, out SocketError error);                         //Recieve the response from the server
   Console.Write(Encoding.ASCII.GetString(buffer), 0, size);                                    //Write the response to console
   client.Close();                                                                              //Close the client
  }
 }
}